logstash-core/lib/logstash/environment.rb

```diff
@@ -3,8 +3,16 @@
 require "logstash/config/cpu_core_strategy"
 require "logstash/settings"
 require "socket"
+require "stud/temporary"
 
 module LogStash
+  # In the event that we're requiring this file without bootstrap/environment.rb
+  if !defined?(LogStash::Environment::LOGSTASH_HOME)
+    module Environment
+      LOGSTASH_HOME = Stud::Temporary.directory("logstash-home")
+      Dir.mkdir(::File.join(LOGSTASH_HOME, "data"))
```

Comment:
@ph it looks like it does here https://github.com/jordansissel/ruby-stud/blob/master/lib/stud/temporary.rb#L47